test/functional/p2p_1p1c_package_relay.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test that package works successfully in a "network" of nodes. Send various packages from different
+nodes on a network in which some nodes have already received some of the transactions (and submitted
+them to mempool, kept them as orphans or rejected them as too-low-feerate transactions). The
+packages should be received and accepted by all transactions on the network.
+"""
+
+from decimal import Decimal
+from test_framework.messages import (
+    CInv,
+    MSG_WTX,
+    msg_inv,
+    msg_tx,
+)
+from test_framework.p2p import (
+    P2PInterface,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_greater_than,
+    create_lots_of_big_transactions,
+    gen_return_txouts,
+)
+from test_framework.wallet import (
+    MiniWallet,
+)
+
+# 1sat/vB feerate denominated in BTC/KvB
+FEERATE_1SAT_VB = Decimal("0.00001000")
+
+class PackageRelayTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 4
```

Comment:
In 19015e3c9aed5709e776ef15bf9e73f126c7ea29: It seems to be using 3 nodes instead of 4. Or am I missing something?

E.g.:

```diff
diff --git a/test/functional/p2p_1p1c_package_relay.py b/test/functional/p2p_1p1c_package_relay.py
index 6a3ba7c3df..dda671dede 100755
--- a/test/functional/p2p_1p1c_package_relay.py
+++ b/test/functional/p2p_1p1c_package_relay.py
@@ -36,7 +36,7 @@ FEERATE_1SAT_VB = Decimal("0.00001000")
 class PackageRelayTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
-        self.num_nodes = 4
+        self.num_nodes = 3
         self.extra_args = [[
             "-datacarriersize=100000",
             "-maxmempool=5",
@@ -85,7 +85,7 @@ class PackageRelayTest(BitcoinTestFramework):
         # Child should already be in orphanage
         self.transactions_to_presend[1] = [high_fee_child["tx"]]
         # Parent would have been previously rejected
-        self.transactions_to_presend[3] = [low_fee_parent["tx"]]
+        self.transactions_to_presend[2] = [low_fee_parent["tx"]]
 
     def test_individual_logic(self):
         node = self.nodes[0]
@@ -213,7 +213,7 @@ class PackageRelayTest(BitcoinTestFramework):
 
         self.log.info("Submit replacement package to node3")
         for package_hex in self.replacement_packages:
-            self.nodes[3].submitpackage(package_hex)
+            self.nodes[2].submitpackage(package_hex)
 
         self.log.info("Wait for mempools to sync")
         self.sync_mempools(timeout=20)
```

Reply:
The idea is this is a "network" test to see that the package propagates across multiple hops. 1 node pre-receives the parent and 1 node pre-receives the child, and there is a node in between.

Reply:
Got it, thank you.